### ISSUE TITLE
Don't show inactive buttons in editor top bar

### ISSUE THREAD

**Issue by passivestar** [godotengine/godot-proposals#8570]
### Describe the project you are working on

Anything in Godot editor

### Describe the problem or limitation you are having in your project

The top panel with buttons to run the project looks busy:

![now](https://github.com/godotengine/godot-proposals/assets/60579014/b868b362-32bf-478f-85fc-3f0dfc9c3b74)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It can easily be decluttered if we don't show inactive buttons

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

This is when the project is not running:

![notrunning](https://github.com/godotengine/godot-proposals/assets/60579014/96e162b5-020e-4ff1-b536-23952e5f14b1)

This is when the project is running:

![running](https://github.com/godotengine/godot-proposals/assets/60579014/cc4e96b4-6d18-4578-a4a9-e738b5b170a5)

Pause and stop buttons can't do anything if the project is not running. Running current/specific scene while the project is already running is technically possible, but I'm not sure if anybody ever pressed those buttons without stopping first. The remote debug button is only active if remote debug is configured, it's kind of an advanced feature and you probably don't need to see an icon for that unless it's enabled in your project

This is a low-hanging fruit to declutter the UI because that's how people usually expect controls like these to work anyway, so there's no need to show them in their inactive state:

<img width="252" alt="Screenshot 2023-12-02 at 18 16 27" src="https://github.com/godotengine/godot-proposals/assets/60579014/5ff8d45c-bd51-47d3-9902-c5605cd0959d">

<img width="695" alt="Screenshot 2023-12-02 at 18 18 41" src="https://github.com/godotengine/godot-proposals/assets/60579014/8d873d53-7351-455d-8af8-7c977baa6676">

Less buttons will make the editor less overwhelming to newcomers and easier on the eyes for existing users

### If this enhancement will not be used often, can it be worked around with a few lines of script?

N/A

### Is there a reason why this should be core and not an add-on in the asset library?

N/A

**Comment by passivestar** [comment]
>  because I don't want buttons to move around

can you elaborate, what button do you want to press twice but won't be able to in this specific case?

**Comment by TheDuriel** [comment]
I don't **want** to be pressing buttons twice. However, accidents happen. And this increases accidents.

A common "operation" that I do IS to: Hit F5 to run the game, realize I missed something, and immediately click stop. Now with this proposal the stop button may jump around. Maybe I'll hit it, maybe not. My memory of where it is has become useless, and I will need to carefully check each buttons position every time.

---

There are other places in the editor where hiding inactive operations can be of benefit. This in my opinion, is not one of them.

**Comment by MBCX** [comment]
I think this could work if this is implemented in the way YouTube does it (for example), where the *stop* and *play* video buttons are in the same location, but they just animate/morph them accordingly.

In this case, if the *play* and *stop* game buttons just switch places (instantly) and they're in the same location, then it could work. However, this **shouldn't be a default.** A tick like "Hide inactive buttons" would have to be implemented because many are used to the current workflow.

**Comment by dragon1freak** [comment]
I can agree to hiding the pause/stop buttons when the project is playing, but hiding UI from the user by default is a slippery slope.  Imo by default you want to disable and tooltip content thats currently inactive so the user knows its there _and_ knows why its disabled, and then we can opt into hiding inactive content in the editor settings somewhere if we really want to.  You want to look at this as someone whos completely new to your application, sure _you_ know what you need to do to get those inactive buttons to show and use them, but does a new person even know that content is there?  

Tl;dr:  Not a bad idea, but make it opt in, not the default

**Comment by AThousandShips** [comment]
In my opinion hiding UI elements on dynamic changes like "running/not running" instead of disabling them is not good UI design, it breaks muscle memory and workflow by moving elements around constantly

**Comment by ztc0611** [comment]
I like this, but clearly from the responses it should be optional if done. I think it should be on by default, though. To me this makes more sense than how it's currently done and leaving it as-is is only good for not breaking existing dev muscle memory.

**Comment by LauraWebdev** [comment]
While I generally agree that button changes are bad for muscle memory, running the game is something I'd consider a "major state change" that would take a second or two to execute anyways.

**Comment by AThousandShips** [comment]
Major state change here would be a change *to the editor* or the settings, running the game isn't a major change, nothing in the editor changes

**Comment by passivestar** [comment]
Ok so will disabling buttons in editor settings be a better proposal to declutter then?

e.g I never press any of these:
![Untitled](https://github.com/godotengine/godot-proposals/assets/60579014/bd348144-b423-47b4-8cb1-5af4154f31da)

**Comment by ztc0611** [comment]
Is it controversial to say at the very least the remote debug button should be hidden if you don't even have `Deploy with Remote Debug` enabled? I can't imagine it's very commonly used besides people who deploy to Android. Sure that would alter someone's muscle memory in the short term if they change from having it off to on, but I just think countering any visual streamlining with "it could affect muscle memory" is its own kind of slippery slope.

**Comment by TheDuriel** [comment]
The remote debug button serves as a method of enabling the functionality. It has a purpose.

If you never want to use it, Editor Feature Profiles would be the way to remove it.

**Comment by ztc0611** [comment]
> The remote debug button serves as a method of enabling the functionality. It has a purpose.

I'm not sure if this is something with macOS or something else, but clicking the button if it is not configured has no effect for me. `Deploy with Remote Debug` is a checkbox item in the Editor menu bar.

<img width="543" alt="image" src="https://github.com/godotengine/godot-proposals/assets/28269330/c0a4c443-fb61-4b81-adf4-44f456b2be0c">
<img width="570" alt="image" src="https://github.com/godotengine/godot-proposals/assets/28269330/819eb322-d85a-40c3-b302-257a740739f4">

**Comment by passivestar** [comment]
Closing this in favor of implementing this as an editor addon, see the discussion [here](https://github.com/godotengine/godot-proposals/issues/8571#issuecomment-1837222452). The relevant proposals to make the addon feasible are https://github.com/godotengine/godot-proposals/issues/1018 and https://github.com/godotengine/godot-proposals/issues/831
